frontend/packages/dev-console/src/actions/modify-application.ts

```diff
@@ -25,8 +25,10 @@ export const ModifyApplication = (kind: K8sKind, obj: K8sResourceKind): KebabOpt
 };
 
 export const EditApplication = (model: K8sKind, obj: K8sResourceKind): KebabOption => {
+  const annotation = obj?.metadata?.annotations?.['openshift.io/generated-by'];
   return {
     label: `Edit ${truncateMiddle(obj.metadata.name, { length: RESOURCE_NAME_TRUNCATE_LENGTH })}`,
+    hidden: !annotation && annotation !== 'OpenShiftWebConsole',
```

Comment:
@vikram-raj @invincibleJai please more careful when reviewing.
```suggestion
    hidden: !annotation || annotation !== 'OpenShiftWebConsole',
```

Reply:
Thanks , missed this. @vikram-raj can you pls update

Reply:
Done.